elisp/intero.el

```diff
@@ -151,7 +151,6 @@ To use this, use the following mode hook:
   (if intero-mode
       (progn
         (intero-flycheck-enable)
-        (add-hook 'completion-at-point-functions 'intero-completion-at-point nil t)
```

Comment:
This gave me pause, because I guess my intention in adding this line to `intero-mode` was to ensure there was always a sensible `completion-at-point` function in place, even in `.hs` source buffers. But we're pretty much forcing `company` on everyone, so that's fine! And I like that the `completion-at-point` function is now clearly named in order to be specific to the REPL.

Reply:
If the repl autocomplete had no completions, I didn't want it to fall through to something else. Reduce inconsistency with the other changes.